src/interaction/pointer.ts

```diff
@@ -28,64 +28,64 @@ function mouse(event: Event, evt: MouseEvent): void {
     let de = document.documentElement;
     let x = "pageX" in evt ? Math.round(evt.pageX) : ("clientX" in evt ? Math.round(evt["clientX"] + de.scrollLeft) : null);
     let y = "pageY" in evt ? Math.round(evt.pageY) : ("clientY" in evt ? Math.round(evt["clientY"] + de.scrollTop) : null);
-    let id = evt.target ? getId(evt.target as Node) : null;
+    let target = evt.target ? evt.target as Node : null;
     event = event === Event.Click && (evt.buttons === 2 || evt.button === 2) ? Event.RightClick : event;
-    handler(event, {target: id, x, y, time: time()});
+    handler({ time: time(), event, data: { target, x, y } });
 }
 
 function touch(event: Event, evt: TouchEvent): void {
     let de = document.documentElement;
     let touches = evt.changedTouches;
-    let id = evt.target ? getId(evt.target as Node) : null;
+    let target = evt.target ? evt.target as Node : null;
     let t = time();
     if (touches) {
         for (let i = 0; i < touches.length; i++) {
             let entry = touches[i];
             let x = "clientX" in entry ? Math.round(entry["clientX"] + de.scrollLeft) : null;
             let y = "clientY" in entry ? Math.round(entry["clientY"] + de.scrollTop) : null;
-            handler(event, {target: id, x, y, time: t});
+            handler({ time: t, event, data: { target, x, y } });
         }
     }
 }
 
-function handler(event: Event, current: PointerData): void {
-    switch (event) {
+function handler(current: PointerState): void {
+    switch (current.event) {
         case Event.MouseMove:
         case Event.MouseWheel:
         case Event.TouchMove:
-            let length = data[event].length;
-            let last = length > 1 ? data[event][length - 2] : null;
-            if (last && similar(last, current)) { data[event].pop(); }
-            data[event].push(current);
+            let length = data.length;
+            let last = length > 1 ? data[length - 2] : null;
+            if (last && similar(last, current)) { data.pop(); }
+            data.push(current);
 
             clearTimeout(timeout);
-            timeout = setTimeout(encode, config.lookahead, event);
+            timeout = setTimeout(process, config.lookahead, current.event);
```

Comment:
Minor: This would be clearer if you passed in just a lambda function and didn't pass the fn and argument(s) separately.